frontend/src/tw-components/Checkbox.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { useId, useRef, useState, useEffect } from "react";
+import { IconCheckboxY, IconCheckboxN } from "assets/images/images";
+import { cn } from "lib/utils";
+
+// Type declaration for props
+interface CheckboxProps {
+  className?: string;
+  defaultChecked?: boolean;
+  disabled?: boolean;
+  label: string;
+  labelHidden?: boolean;
+  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
+}
+
+function Checkbox({
+  defaultChecked = false,
+  disabled = false,
+  labelHidden = false,
+  ...props
+}: CheckboxProps) {
+  const checkboxId = useId();
+  const checkboxRef = useRef<HTMLInputElement>(null);
+  const [isChecked, setIsChecked] = useState(defaultChecked);
+
+  useEffect(() => {
+    if (checkboxRef.current) {
+      checkboxRef.current.checked = isChecked;
+    }
+  }, [isChecked]);
+
+  function handleChange(e: React.ChangeEvent<HTMLInputElement>) {
+    if (!disabled) {
+      setIsChecked(e.target.checked);
+    }
+    if (props.onChange) {
+      props.onChange(e);
+    }
+  }
+
+  return (
+    <div className={cn("relative flex items-center", props.className)}>
+      <input
+        id={checkboxId}
+        type="checkbox"
+        defaultChecked={defaultChecked}
+        disabled={disabled}
+        ref={checkboxRef}
+        onChange={handleChange}
+        className="peer sr-only"
+      />
+
+      {/* Circle Effect */}
+      <span
+        className={cn(
```

Comment:
```suggestion
        className="absolute h-10 w-10 rounded-full opacity-0 peer-focus:bg-blue-dark peer-focus:opacity-[.16] peer-active:bg-blue-dark peer-active:opacity-[.32]"
```

We don't need `cn` here since the classnames are static and dont change depending on props or state